docs/src/main/tut/FeatureOverview.md

```diff
@@ -215,6 +215,190 @@ case class PriceInfo3(ratio: Int, price: Double) // ratio should be Double
+
+```tut:book
+aptTypedDs2.dropTupled('price): TypedDataset[(String,Int)]
+```
+
+To drop a column and specify a new schema use `drop()`.
+
+```tut:book
+case class CityBeds(city: String, bedrooms: Int)
+val cityBeds: TypedDataset[CityBeds] = aptTypedDs2.drop[CityBeds] 
+```
+
+Often, you want to replace an existing column with a new value.
+ 
+```tut:book
+val inflation = aptTypedDs2.withColumnReplaced('price, aptTypedDs2('price) * 2)
+ 
+inflation.show(2).run()
+```
+
+Or use a literal instead.
+
+```tut:book
+import frameless.functions.lit
+aptTypedDs2.withColumnReplaced('price, lit(0.001)) 
+```
+
+Adding a column using `withColumnTupled()` results in a tupled-based schema.
+
+```tut:book
+aptTypedDs2.withColumnTupled(lit(Array("a","b","c"))).show(2).run()
+```
+
+Similarly, `withColumn()` adds a column and explicitly expects a schema for the result.
+
+```tut:book
+case class CityBedsOther(city: String, bedrooms: Int, other: List[String])
+
+cityBeds.
+   withColumn[CityBedsOther](lit(List("a","b","c"))).
+   show(1).run()
+```
+
+To conditionally change a column use the `when/otherwise` operation. 
+
+```tut:book
+import frameless.functions.nonAggregate.when
+aptTypedDs2.withColumnTupled(
+   when(aptTypedDs2('city) === "Paris", aptTypedDs2('price)).
+   when(aptTypedDs2('city) === "Lyon", lit(1.1)).
+   otherwise(lit(0.0))).show(8).run()
+```
+
+A simple way to add a column without loosing important schema information is
+to project the entire source schema into a single column. 
+
+```tut:book
+val c = cityBeds.select(cityBeds.asCol, lit(List("a","b","c")))
+c.show(1).run()
+```
+
+Using `select()` and `asCol()`, compared to using `withColumn()`, avoids the 
+need of an extra `case class` to define the result schema.
+
+To access nested columns, use the `colMany()` method. 
+
+```tut:book
+c.select(c.colMany('_1, 'city), c('_2)).show(2).run()
+```
+
+
+### Working with collections
+
+
+```tut:book
+import frameless.functions._
+import frameless.functions.nonAggregate._
+```
+
+```tut:book
+val t = cityRatio.select(cityRatio('city), lit(List("abc","c","d")))
+t.withColumnTupled(
+   arrayContains(t('_2), "abc")
+).show(1).run()
+```
+
+If accidentally you apply a collection function on a column that is not a collection,
+you get a compilation error.
+
+```tut:book:fail
+t.withColumnTupled(
+   arrayContains(t('_1), "abc")
+)
+```
+
+
+### Collecting data to the driver
+
+In Frameless all Spark actions (such as `collect()`) are safe.
+
+Take the first element from a dataset (if the dataset is empty return `None`).
+
+```tut:book
+cityBeds.headOption.run()
+```
+
+Take the first `n` elements.
+
+```tut:book
+cityBeds.take(2).run()
+```
+
+```tut:book
+cityBeds.head(3).run()
+```
+
+```tut:book
+cityBeds.limit(4).collect().run()
+```
+
+## Sorting columns
+
+
+Only column types that can be sorted are allowed to be selected for sorting. 
+
+```tut:book
+aptTypedDs.orderBy(aptTypedDs('city).asc).show(2).run()
```

Comment:
I had it like that but implicit with poly didn’t work in tut. Not sure why ... I will give it another shot

Reply:
Hmm you're right. It seems that explicitly calling the `apply` with the column type works but otherwise it fails to infer the sort. E.g.:
```scala
aptTypedDs.orderBy(aptTypedDs[String]('city).asc).show(2).run()
```
That's fairly annoying.